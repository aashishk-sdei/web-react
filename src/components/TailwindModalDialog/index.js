import React from "react";
import Button from "./../Button";
import TailwindModal from "./../TailwindModal";

const TailwindModalDialog = ({
    showModal,
    setShowModal,
    handleAction,
    content,
    title,
    loading
}) => {
    return <TailwindModal showModal={showModal}   
    innerClasses = "max-w-edit-search-modal-w modal-xs" 
    setShowModal = {setShowModal}
    title={title}
    content={
        <><p>{content}</p>
        <div className="flex justify-end mt-10">
            <Button 
            size="large"
            type="default"
            handleClick={(e) => {
                e.preventDefault();
                setShowModal(false);
            }}
            fontWeight="medium"
            title="Cancel"/>
      <div className="ml-2">
      <Button 
            handleClick={(e) => {
                e.preventDefault();
                handleAction();
            }}
            loading={loading}
            size="large"
            type="danger"
            fontWeight="medium"
            title="Delete"/>
      </div>
            </div>
        </>
    }
    />
}
export default TailwindModalDialog