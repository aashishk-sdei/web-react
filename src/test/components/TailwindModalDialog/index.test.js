import React from "react";
import { mount } from "enzyme";
import TailwindModalDialog from "../../../components/TailwindModalDialog";

describe('Test TailwindModalDialog component with props', () => {
    const mockCallBack = jest.fn(()=> true);
    const mockCallBackStopProp = jest.fn(()=> true);
    let modal;
    beforeEach(() => {
      modal = mount(<TailwindModalDialog showModal = {true}
        setShowModal = {mockCallBack}
        clickAwayClose = {false}
        title = {"Test TailwindModalDialog"}
        content = {"Test TailwindModalDialog Content"}/>);
    });
    it('match snapshot', () => {
      expect(modal).toMatchSnapshot();
    });
    it('match showModal props is true', () => {
      expect(modal.props().showModal).toEqual(true);
    });
    it('match subTitle props', () => {
      modal.setProps({ subTitle: "test Here" });
      expect(modal.props().subTitle).toEqual("test Here");
    });
    it('match showModal props is false', () => {
      modal.setProps({ showModal: false });
      expect(modal.props().showModal).toEqual(false);
    });
    it('match showClose props is false', () => {
      modal.setProps({ showClose: false });
      expect(modal.props().showClose).toEqual(false);
    });
    it('match for close button click', () => {
      modal.find('button.outline-none').simulate('click', {
        stopPropagation: jest.fn()
      });
      expect(mockCallBack.mock.calls.length).toEqual(1);
    })
    it('match for stop on modal click', () => {
      modal.find('#twModal').simulate('click', {
        stopPropagation: mockCallBackStopProp
      });
      expect(mockCallBackStopProp.mock.calls.length).toEqual(1);
    })
    afterEach(() => {
      modal.unmount()
    });
});