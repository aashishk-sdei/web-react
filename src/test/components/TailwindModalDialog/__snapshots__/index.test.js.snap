// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TailwindModalDialog component with props match snapshot 1`] = `ReactWrapper {}`;
